docs/how_to_contribute/math/how_to_add_prime_field.md

```diff
@@ -0,0 +1,409 @@
+  // MultiplicativeSemigroup methods
+  constexpr PrimeField& MulInPlace(const PrimeField& other) {
+    PrimeField ret;
+    Fr_rawMMul(ret.value_.limbs, value_.limbs, other.value_.limbs);
+    *this = ret;
+    return *this;
+  }
+
+  constexpr PrimeField& SquareInPlace() {
+    PrimeField ret;
+    Fr_rawMSquare(ret.value_.limbs, value_.limbs);
+    *this = ret;
+    return *this;
+  }
+
+  // MultiplicativeGroup methods
+  PrimeField& DivInPlace(const PrimeField& other) {
+    return MulInPlace(other.Inverse());
+  }
+
+  constexpr PrimeField& InverseInPlace() {
+    value_ = value_.template MontgomeryInverse<Config::kModulusHasSpareBit>(
+        Config::kModulus, Config::kMontgomeryR2);
+    return *this;
+  }
+
+ private:
+  BigInt<N> value_;
+};
+
+}  // namespace tachyon::math
+
+#endif  // TACHYON_MATH_FINITE_FIELDS_MERSENNE_31_PRIME_PRIME_FIELD_MERSENNE_31_H_
+
+```
+
+## 3. Add unit test
+
+Finally, ensure your Prime Field works well by incorporating it into the `prime_field_generator_unittest.cc`.
+
+```cpp
+#include "gtest/gtest.h"
+
+#include "tachyon/math/elliptic_curves/bls12/bls12_381/fq.h"
+#include "tachyon/math/elliptic_curves/bls12/bls12_381/fr.h"
+#include "tachyon/math/elliptic_curves/bn/bn254/fq.h"
+#include "tachyon/math/elliptic_curves/bn/bn254/fr.h"
+#include "tachyon/math/elliptic_curves/secp/secp256k1/fq.h"
+#include "tachyon/math/elliptic_curves/secp/secp256k1/fr.h"
+#include "tachyon/math/finite_fields/goldilocks_prime/goldilocks.h"
+// Include the generated code. 
+#include "tachyon/math/finite_fields/mersenne_31_prime/mersenne_31.h"
+
+namespace tachyon::math {
+
+namespace {
+template <typename PrimeField>
+class PrimeFieldGeneratorTest : public testing::Test {
+ public:
+  static void SetUpTestSuite() { PrimeField::Init(); }
+};
+
+}  // namespace
+
+// Add the new Prime Field type to test
+using PrimeFieldTypes =
+    testing::Types<bls12_381::Fq, bls12_381::Fr, bn254::Fq, bn254::Fr,
+                   secp256k1::Fr, secp256k1::Fq, Goldilocks, Mersenne31>;
+
+TYPED_TEST_SUITE(PrimeFieldGeneratorTest, PrimeFieldTypes);
+
+TYPED_TEST(PrimeFieldGeneratorTest, FromString) {
+  using PrimeField = TypeParam;
+  EXPECT_EQ(PrimeField::FromDecString("3"), PrimeField(3));
+  EXPECT_EQ(PrimeField::FromHexString("0x3"), PrimeField(3));
+}
+
+TYPED_TEST(PrimeFieldGeneratorTest, ToString) {
+  using PrimeField = TypeParam;
+  PrimeField f(3);
+
+  EXPECT_EQ(f.ToString(), "3");
+  EXPECT_EQ(f.ToHexString(), "0x3");
+}
+
+TYPED_TEST(PrimeFieldGeneratorTest, Zero) {
+  using PrimeField = TypeParam;
+  EXPECT_TRUE(PrimeField::Zero().IsZero());
+  EXPECT_FALSE(PrimeField::One().IsZero());
+}
+
+TYPED_TEST(PrimeFieldGeneratorTest, One) {
+  using PrimeField = TypeParam;
+  EXPECT_TRUE(PrimeField::One().IsOne());
+  EXPECT_FALSE(PrimeField::Zero().IsOne());
+  EXPECT_EQ(PrimeField::Config::kOne, PrimeField(1).ToMontgomery());
+}
+
+// More tests ...
+
+}  // namespace tachyon::math
+
+```
```

Comment:
something along these lines? I think including a large part of the file is not really needed...
```suggestion
```cpp
// prime_field_generator_unittest.cc
...
#include "tachyon/math/finite_fields/goldilocks_prime/goldilocks.h"
#include "tachyon/math/finite_fields/mersenne_31_prime/mersenne_31.h"
// ADD NEW PRIME FIELD HEADER FILE HERE
// #include "tachyon/math/finite_fields/..."
...
// ADD NEW PRIME FIELD NAME HERE
using PrimeFieldTypes =
    testing::Types<bls12_381::Fq, bls12_381::Fr, bn254::Fq, bn254::Fr,
                   secp256k1::Fr, secp256k1::Fq, Goldilocks, Mersenne31 /*, NEW PRIME FIELD*/>;
...
```